### PR TITLE
Build for either ubuntu or alpine with Docker

### PR DESCRIPTION
**Pull request by RichardoC**
Also added some convenience for those using [nerdctl](https://github.com/containerd/nerdctl/releases) due to the docker desktop license changes.

### CODE REVIEW

**Comment by RichardoC** [comment]
Should mean noone else needs to open the equivalent of https://github.com/rui314/mold/issues/187

**Comment by rui314** [comment]
This may be a dumb question, but what is the point of allowing Alpine to create a mold executable? Is that better than Ubuntu in some way?

Please add a `signed-off-by` line to your commit as explained in https://github.com/rui314/mold/blob/main/CONTRIBUTING.md.

**Comment by RichardoC** [comment]
Ubuntu uses glibc and Alpine uses MUSL. It is possible to compile for MUSL on Ubuntu but it's easier to just build it on Alpine.
I've fixed my commits now, sorry about that! For anyone who does the same thing, here's a simple script to fix them
```bash
export SIGNOFF="signed-off-by: $(git config --get user.name) <$(git config --get user.email)>"
git filter-branch -f --msg-filter     "git interpret-trailers --trailer \"$SIGNOFF\""      HEAD~10..HEAD
git push --force
```

File: test/elf/bug179.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+export LANG=
+set -e
+cd $(dirname $0)
+mold=`pwd`/../../mold
```
Comment (RichardoC) [inline test/elf/bug179.sh]: If you're always running this in a git repo, you could use 
```bash
export ROOT=$(git rev-parse --show-toplevel | tr -d '\n')
mold="$ROOT/mold"
```

**Comment by rui314** [comment]
So I guess your intention is to get rid of glibc from the statically-linked mold executable and use musl instead. Is this correct? If so, how about also removing the dependency to GNU's libstdc++ and switch to LLVM's libc++? If it's doable, and if it doesn't negatively impact mold's performance, I'd do that.

After that, we can simply replace the current `build-static.sh` with your Alpine's one.

**Comment by rui314** [comment]
It looks like your pull request contains my commits. Please rebase so that this PR consists only of your changes.

**Review by sicherha — COMMENTED** [review]
Could you explain what these additions to `CPPFLAGS` and `LIBS` are supposed to achieve? The comment says something about stuff being installed to `/lib` instead of `/usr/local/lib`, but that doesn't seem to have any relation to what is actually being done:
* Add `/lib` to the preprocessor's include path
* Add the root directory to the linker's library path

Unless Alpine has an incredibly weird filesystem layout that I'm unaware of, I don't think these changes are actually needed.

**Comment by RichardoC** [comment]
Abandoning in favour of https://github.com/rui314/mold/pull/166
Thanks for the reviews everyone, that's a much better solution than this PR

**Pull request closed without merging**